hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
Why does it matter what the destination is?  It doesn't matter what x64 or ppc64 does when we're talking about implementing a vasm instruction.  In vasm, the destination of a `movb{}` is a Vreg8, which means that a `movb{}` makes no guarantees about the upper bits of its destination (assuming the Vreg8 is backed by a regular 64-bit physical register).

If you need to copy 8 bits and zero all the upper bits, you need to lower to `movzbq{s, tmp}; movtqb{tmp, d}` instead of to `movb{s, d}`.  Is that what you want?